sfcc_cartridge/int_fast_sfra/cartridge/static/default/js/fastCart.js

```diff
@@ -1,44 +1,66 @@
 'use strict';
+const myTimeout = setTimeout(initCartFastBtn, 2000);
 
-$(document).ready(function () {
-    var fastButtonExist = setInterval(function() {
-        if ($('fast-checkout-cart-button').length) {
-            var fast = new Fast();
-            var checkoutFastJsButton = $('fast-checkout-cart-button');
-
-            checkoutFastJsButton.click(function(event) {
-                // This is not the final file is in WIP for future changes
-                if (event.target.closest('.cart-page')) {
-                    fast.addEventListener("user_event", function(event) {
-            
-                        var cartClearingEvents = [
-                            "Checkout - Order Created",
-                            "Checkout - Order Updated",
-                            "Checkout - Order Completed",
-                        ];
+function initCartFastBtn() {
+    
+    $(document).ready(function () {
+        // following code is for cart page fast checkout button to check checkout events
+        var cartPageCheckoutButton = $('.cart-page fast-checkout-cart-button');
+        
+        if (cartPageCheckoutButton.length > 0) {
+            cartPageCheckoutButton.click(function(){
+                var fast = new Fast();
                 
-                        if (cartClearingEvents.includes(event.name)) {
+                fast.addEventListener("user_event", function(event) {
+                    
+                    var cartClearingEvents = [
```

Comment:
CartClearingEvents defined multiple times (line #41). please define once and reuse across methods where needed